### PR TITLE
New enterprise flow

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Change the internal process for utilizing enterprise modules in our codebase.

### Why was this change needed?

So we could use the enterprise module.

### CODE REVIEW

**Comment by Cliftonz** [comment]
@djabarovgeorge Can you add more context to this?

File: .source
Comment (BiswaViraj) [inline .source:1]: why do we need this `.source` ?

Comment (djabarovgeorge) [reply .source:1]: This is the new submodule link, instead of the old one 'enterprise'

File: .github/workflows/reusable-api-e2e.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ on:
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   check_submodule_token:
+    if: ${{ inputs.ee }}
```
Comment (djabarovgeorge) [inline .github/workflows/reusable-api-e2e.yml:30]: Run this check only if the matrix name is `'novu/api-ee'` otherwise, it is not needed.

**Review by LetItRock — APPROVED** [review]
looks good to me ⭐

**Pull request merged**